internal/controller/apiextensions/claim/syncer_ssa.go

```diff
@@ -0,0 +1,292 @@
+	if !ok {
+		return errors.New(errUnsupportedClaimSpec)
+	}
+
+	// Propagate the claim's spec (minus well known fields) to the XR's spec.
+	xrPatch.Object["spec"] = withoutKeys(cmSpec, xcrd.GetPropFields(wellKnownClaimFields)...)
+
+	// We overwrite the entire XR spec above, so we wait until this point to set
+	// the claim reference.
+	xrPatch.SetClaimReference(cm.GetReference())
+
+	// Below this point we're syncing XR -> claim.
+
+	// Bind the claim to the XR. If this is a new XR it's important that we
+	// apply the claim before we create it. This ensures we don't leak an XR. We
+	// could leak an XR if we created an XR then crashed before saving a
+	// reference to it. We'd create another XR on the next reconcile.
+	cm.SetResourceReference(meta.ReferenceTo(xrPatch, xrPatch.GroupVersionKind()))
+
+	// Propagate the actual external name back from the composite to the
+	// claim if it's set. The name we're propagating here will may be a name
+	// the XR must enforce (i.e. overriding any requested by the claim) but
+	// will often actually just be propagating back a name that was already
+	// propagated forward from the claim to the XR during the
+	// preceding configure phase.
+	if en := meta.GetExternalName(xr); en != "" {
+		meta.SetExternalName(cm, en)
+	}
+
+	// Propagate composition ref from the XR if the claim doesn't have an
+	// opinion. Composition and revision selectors only propagate from claim ->
+	// XR. When a claim has selectors **and no reference** the flow should be:
+	//
+	// 1. Claim controller propagates selectors claim -> XR.
+	// 2. XR controller uses selectors to set XR's composition ref.
+	// 3. Claim controller propagates ref XR -> claim.
+	//
+	// When a claim sets a composition ref, it supercedes selectors. It should
+	// only be propagated claim -> XR.
+	if ref := xr.GetCompositionReference(); ref != nil && cm.GetCompositionReference() == nil {
+		cm.SetCompositionReference(ref)
+	}
+
+	// Propagate composition revision ref from the XR if the update policy is
+	// automatic. When the update policy is automatic the XR controller is
+	// authoritative for this field. It will update the XR's ref as new
+	// revisions become available, and we want to propgate the ref XR -> claim.
+	if p := xr.GetCompositionUpdatePolicy(); p != nil && *p == xpv1.UpdateAutomatic && xr.GetCompositionRevisionReference() != nil {
+		cm.SetCompositionRevisionReference(xr.GetCompositionRevisionReference())
+	}
+
+	// It's important that we update the claim before we apply the XR, to make
+	// sure the claim's resourceRef points to the XR before we create the XR.
+	// Otherwise we risk leaking an XR.
+	//
+	// It's also important that the API server will reject this update if we're
+	// reconciling an old claim, e.g. due to a stale cache. It's possible that
+	// we're seeing an old version without a resourceRef set, but in reality an
+	// XR has already been created. We don't want to leak it and create another.
```

Comment:
This comment explains why we don't SSA the claim anymore.

To be honest, I think adopting the practices suggested by https://github.com/crossplane/crossplane/issues/4581 means this controller doesn't gain a lot from SSA-ing the XR, either.

We no longer have any objects or arrays we need to worry about merging. We mostly just overwrite the entire XR spec and entire claim status. So we're not getting any of the benefits of server-side merging. Mostly we're just getting marginally simpler code in that we can just build and send an SSA patch for the XR, rather than having to do  a `Get`, mutate, `Update`.